### DISCUSSION TITLE
Intercepting Modal

### DISCUSSION THREAD

**Discussion by ivenzdev** [vercel/next.js#59946] [App Router]
https://github.com/IVEN21/Modal-Intercept

If I would implement a functionality that when I open modal (/photo/2), I can navigate to (/photo/3). In this case, how would I close the modal when I'm in photos 3 that navigated from photos 2?

Any help is very much appreciated!

**Comment by bchmnn** [comment]
Within your modal, you could navigate using:
```
<Link href={`/photos/${id + 1}`} replace>{...}</Link>
```
Note that the `replace` flag's functionality is described as follows:
```
Replace the current `history` state instead of adding a new url into the stack.
```

To close the modal, you can now make use of the `'next/navigation'` router:
```
<button onClick={() => router.back()}>{...}</button>
```

Since we replaced the history entries of previous photos with entries of the current photos, the previous entry is the route from which the modal was opened.

**Reply by ivenzdev** [reply]
But going back and forth on the history is a core functionality in my app, when replacing the history, the history resulted inconsistent.

**Reply by ananuness** [reply]
@ivenzdev what about using the same logic, but switching `.replace()` to `.push()`?

**Reply by ivenzdev** [reply]
I need a logic that can close the modal all at once, I need to use router.back(), but it only can go back once.
